src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/AmbientAuthenticationState.cs

```diff
@@ -8,7 +8,7 @@ namespace NuGet.Protocol
     /// </summary>
     public class AmbientAuthenticationState
     {
-        internal const int MaxAuthRetries = 3;
+        internal const int MaxAuthRetries = 4;
```

Comment:
Added an additional retry, because the DefaultCredentialsCredentialProvider "consumes" a retry before falling back to prompting at the console